File: articles/institutional-enforcement-behavioral-incentives/cpp/institutional_enforcement_score.cpp
#include <iostream>

struct InstitutionalEnforcementCase {
    double monitoring_quality;
    double legitimacy;
    double incentive_alignment;
    double sanction_credibility;
    double information_quality;
    double adaptive_learning;
    double accountability_reach;
    double compliance_burden;
    double selective_enforcement;
    double evasion_pressure;
    double hypocrisy_visibility;
    double defensive_compliance;
};

double institutional_enforcement_score_raw(const InstitutionalEnforcementCase& x) {
    return
        0.13 * x.monitoring_quality +
        0.13 * x.legitimacy +
        0.12 * x.incentive_alignment +
        0.12 * x.sanction_credibility +
        0.13 * x.information_quality +
        0.11 * x.adaptive_learning +
        0.10 * x.accountability_reach -
        0.08 * x.compliance_burden -
        0.08 * x.selective_enforcement -
        0.12 * x.evasion_pressure -
        0.06 * x.hypocrisy_visibility -
        0.06 * x.defensive_compliance;
}

int main() {
    InstitutionalEnforcementCase demo {82, 76, 74, 72, 80, 75, 70, 30, 24, 22, 18, 20};
    std::cout << "Institutional enforcement raw score: "
              << institutional_enforcement_score_raw(demo) << std::endl;
    return 0;
}

File: articles/institutional-enforcement-behavioral-incentives/fortran/institutional_enforcement_score.f90
program institutional_enforcement_score
  implicit none

  real :: monitoring_quality, legitimacy, incentive_alignment, sanction_credibility
  real :: information_quality, adaptive_learning, accountability_reach
  real :: compliance_burden, selective_enforcement, evasion_pressure
  real :: hypocrisy_visibility, defensive_compliance, score

  monitoring_quality = 82.0
  legitimacy = 76.0
  incentive_alignment = 74.0
  sanction_credibility = 72.0
  information_quality = 80.0
  adaptive_learning = 75.0
  accountability_reach = 70.0
  compliance_burden = 30.0
  selective_enforcement = 24.0
  evasion_pressure = 22.0
  hypocrisy_visibility = 18.0
  defensive_compliance = 20.0

  score = 0.13 * monitoring_quality + &
          0.13 * legitimacy + &
          0.12 * incentive_alignment + &
          0.12 * sanction_credibility + &
          0.13 * information_quality + &
          0.11 * adaptive_learning + &
          0.10 * accountability_reach - &
          0.08 * compliance_burden - &
          0.08 * selective_enforcement - &
          0.12 * evasion_pressure - &
          0.06 * hypocrisy_visibility - &
          0.06 * defensive_compliance

  print *, "Institutional enforcement raw score:", score
end program institutional_enforcement_score
